lib/game.rb

```diff
@@ -0,0 +1,135 @@
+require_relative '../lib/player'
+
+class Game
+  WINNING_SET = [
+    [1, 2, 3],
+    [4, 5, 6],
+    [7, 8, 9],
+    [1, 4, 7],
+    [2, 5, 8],
+    [3, 6, 9],
+    [1, 5, 9],
+    [3, 5, 7]
+  ].freeze
+
+  attr_reader :player1, :player2, :game_arr
+
+  def initialize
+    @player1 = Player.new('', 'X')
+
+    @player2 = Player.new('', 'O')
+
+    @game_arr = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+  end
+
+  def start_game
+    i = 0
+
+    player_counter = 0
+
+    win = false
+
+    while i < 9 && win == false
+
+      valid_pos = false
+
+      input_req = player_counter.even? ? @player1.name : @player2.name
+
+      ask_input(input_req)
+
+      until valid_pos == true
+
+        user_in = recieve_turns
+
+        if validate_move(user_in, @game_arr)
+
+          if player_counter.even?
+
+            @player1.choice_array.push(user_in + 1)
+
+            @game_arr[user_in] = @player1.key
+
+            if check_win?(WINNING_SET, @player1.choice_array)
+
+              announce_winner(@player1.name)
+
+              win = true
+
+            end
+
+          else
+
+            @player2.choice_array.push(user_in + 1)
+
+            @game_arr[user_in] = @player2.key
+
+            if check_win?(WINNING_SET, @player2.choice_array)
+
+              announce_winner(@player2.name)
+
+              win = true
+
+            end
+
+          end
+
+          valid_pos = true
+
+        else
+
+          position_error
+
+        end
+
+      end
+
+      display_board(@game_arr)
+
+      i += 1
+
+      player_counter += 1
+
+    end
+    draw unless win == true
+  end
```

Comment:
**OPTIONAL**
That snippet of code sounds good
```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move
  
  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```
But as I said up there, in the `bin/main.rb`